content/200-concepts/100-components/01-prisma-schema/07-names-in-underlying-database.mdx

```diff
@@ -225,3 +225,77 @@ const user = await prisma.user.findUnique({
   },
 })
 ```
+
+#### Filter on composite types
+
+<Admonition type="warning">
+
+Composite types are only available in MongoDB.
+
+</Admonition>
+
+In Prisma versions 4.3.0 and later, you can search on a multi-field constraint that contains composite types. For example, the following schema has a `Product` model, an embedded `Photo` composite type, and a multi-field unique constraint on the product `name` and photo `url`:
+
+```prisma file=schema.prisma
+model Product {
+  id          String  @id @default(auto()) @map("_id") @db.ObjectId
+  name        String  @unique
+  description String
+  photos      Photo[]
+
+  @@unique([name, photos.url]) // multi-field unique constraint
+}
+
+type Photo {
+  caption  String
+  url      String
+}
+```
+
+In this case , the default API to select on the constraint uses a generated combination of the model field name, and the type name and field. For example, the following query searches on `name_photo_url` to get the `Product` record with a given `name`, and containing a `Photo` composite type with a given `url`:
+
+```ts
+const product = await prisma.product.findUnique({
+  where: {
+    name_photo_url: {
+      name: 'Forest Runners',
+      photo: {
+        url: '1.jpg',
+      },
+    },
+  },
+})
+```
+
+To change the default name, use the `name` argument in the `@@unique` attribute. For example, if you change the name to `nameandurl` in the schema as follows:
+
+```prisma file=schema.prisma
+model Product {
+  id          String  @id @default(auto()) @map("_id") @db.ObjectId
+  name        String  @unique
+  description String
+  photos      Photo[]
+
+  @@unique([name, photos.url], name: "nameandurl") // multi-field unique constraint
```

Comment:
```suggestion
  @@unique([name, photos.url], name: "nameandurl") 
```

 I think we should remove this comment because it's not just 'multi field' and users may think they can only do multi.